hypothesis-python/src/hypothesis/strategies/_internal/datetime.py

```diff
@@ -25,9 +27,30 @@
     deprecated_posargs,
     just,
     none,
+    sampled_from,
 )
 from hypothesis.strategies._internal.strategies import SearchStrategy
 
+# These standard-library modules are required for the timezones() and
+# timezone_keys() strategies, but not present in older versions of Python.
+# We therefore try to import them here, but only raise errors recommending
+# `pip install hypothesis[zoneinfo]` to install the backports (if needed)
+# when those strategies are actually used.
+try:
+    import importlib.resources as importlib_resources
+except ImportError:
+    try:
+        import importlib_resources  # type: ignore
+    except ImportError:
+        importlib_resources = None  # type: ignore
+try:
+    import zoneinfo
+except ImportError:
+    try:
+        from backports import zoneinfo  # type: ignore
```

Comment:
This broke `backports.zoneinfo`'s test suite (mainly because `backports.zoneinfo`'s test suite relies on importing `backports.zoneinfo` in its own weird way.

Not that you need to change anything, just thought it's kinda funny.